### PR TITLE
[driver] Buildx implem

### PR DESCRIPTION
**Pull request by redref**

### CODE REVIEW

File: pkr/cli/log.py [hunk 1]
```diff
@@ -22,6 +23,10 @@ def write(msg, add_return=True, error=False):
     fd.flush()
 
 
+def write_fn(fn, msg):
+    os.write(fn, msg.encode("utf-8"))
```
Comment (Kimamisa) [inline pkr/cli/log.py:27]: I doubt black will tolerate a 2 chars var name

File: pkr/driver/buildx.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
+# -*- coding: utf-8 -*-
+# Copyright© 1986-2020 Altair Engineering Inc.
```
Comment (Kimamisa) [inline pkr/driver/buildx.py:2]: 2021 ?

File: pkr/driver/buildx.py [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+                        executor.submit(
+                            *self._build_image(
+                                service, tag, verbose, logfile, nocache, no_rebuild, True
+                            )
+                        )
+                    )
+            for future in futures:
+                future.result(timeout=1800)
+        else:
+            if len(services) > 1:
```
Comment (Kimamisa) [inline pkr/driver/buildx.py:174]: > 0 ?

File: pkr/driver/buildx.py [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+        image_name = self.make_image_name(service, tag)
+
+        dockerfile = self.kard.env.get_container(service).get("dockerfile")
+        if not dockerfile:
+            return
+
+        if no_rebuild:
+            image = len(self.docker.images(image_name)) == 1
+
+        if not no_rebuild or image is False:
```
Comment (Kimamisa) [inline pkr/driver/buildx.py:213]: image might not be defined if no_rebuild is False

File: pkr/cli/log.py (hunk 1 above)
Comment (redref) [reply pkr/cli/log.py:27]: It is a pylint warning yes. But not in the mode we currently use.

File: pkr/driver/buildx.py (hunk 4 above)
Comment (redref) [reply pkr/driver/buildx.py:213]: That's intended.

File: pkr/driver/buildx.py (hunk 2 above)
Comment (redref) [reply pkr/driver/buildx.py:2]: Done

File: pkr/driver/buildx.py (hunk 3 above)
Comment (redref) [reply pkr/driver/buildx.py]: Done, but was coming from docker driver: -)

**Pull request merged**